recipe/meta.yaml

```diff
@@ -1,13 +1,13 @@
 {% set name = "delta" %}
-{% set version = "v2.0_gamma2" %}
+{% set version = "2.0b3" %}
 
 package:
   name: {{ name|lower }}2
   version: {{ version }}
 
 source:
-  url: https://gitlab.com/dunloplab/{{ name }}/-/archive/master/{{ name }}-master.tar.gz
-  sha256: 4bb051fffd668ecc686e0a9a78b93c6fe647878eaeefbb91c4c1505a31c1a5dc 
+  url: https://gitlab.com/dunloplab/{{ name }}/-/archive/main/{{ name }}-main.tar.gz
```

Comment:
I think you're not getting the correct version because you're not adding it in the link here. This should be https://gitlab.com/dunloplab/delta/-/archive/2.0-gamma3/delta-2.0-gamma3.tar.gz and the version should be `2.0-gamma3`.

Reply:
So we made a release called "2.0-gamma3" that was associated with main branch on our gitlab. I had orginally set it up to be the link you described above. I had {{ version }} instead of the main/master.

However, the issue is that conda-forge won't accept that version name because it does not follow PEP440. So I am trying to change the version name without doing another release on gitlab. I thought I could just link it to the main branch instead of the specific release. But it seems like conda-forge knows that it is associated with the release.

Reply:
I just tried to link the url to our dev branch because I know that's not associated with any release. However, I'm still getting the same issue (It won't recognize the new version name in the meta.yml file). Is the only resolution to create a new relase on gitlab?

Reply:
Also thank you so much for responding

Reply:
Disclaimer: I have not checked this in-depth but I believe the version mismatch comes from the fact that you're still using the version in the actual package (`VERSION = "2.0_gamma3"` in _setup.py_). So when you install it, pip will select it from there, not from the one you specify in the _meta.yaml_. 

If it's not too much of a hassle, I would create a new release.

Reply:
I really appreicate your response. This makes a lot of sense that it was grabbing the `VERSION` from the _setup.py_. 

However, I just saw your edited comment now. We ended up making a new commit on the main branch changing the version to ( `VERSION = "2.0b3"`) with in _setup.py_. Then, we made a new release `2.0b3` and I properly linked the git url in the _meta.yml_ file. I will definitely look into the patches for the futuer though.